### PR TITLE
Adding support for export managed project back to manifest

### PR DESCRIPTION
**Pull request by kpavel**
Introducing command to export managed project into manifest file using
`wskdeploy export -p <MANAGED_PROJECT_NAME>`-m <TARGET_MANIFEST_FILENAME>
Project assets will be stored in specified manifest file and all the
actions functions code will be stored in the same directory in respective
files.
All the exported assets (packages/actions...etc) must be managed by same
project via annotations in a regular wskdeploy way.
Currently there may be an issue with multiple managed packages under same
project, as triggers and rules currently placed under packages in the
wskdeploy manifest model, instead of top level similar to OW model

### CODE REVIEW

File: cmd/export.go [hunk 1]
```diff
@@ -0,0 +1,266 @@
...
+				if maniyaml.Packages == nil {
+					maniyaml.Packages = make(map[string]parsers.Package)
+				}
+
+				maniyaml.Packages[pkg.Name] = *maniyaml.ComposeParsersPackage(pkg)
+				// TODO: throw if there more than single package managed by project
+				// currently will be a mess because triggers and rules managed under packages
+				// instead of the top level (similar to OW model)
+				if len(maniyaml.Packages) > 1 {
+					return errors.New("currently can't work with more than one package managed by one project")
```
Comment (pritidesai) [inline cmd/export.go]: we generally define errors under `wskderrors` with a translatable strings in `en_US.all.json` and `i18n_ids.go`, please open an issue if this PR is merged as is to change using `errors.new()` to `wskderrors`, we can take care of it.

File: cmd/export.go [hunk 2]
```diff
@@ -0,0 +1,266 @@
...
+
+	parsers.Write(maniyaml, utils.Flags.ManifestPath)
+	fmt.Println("Manifest exported to: " + utils.Flags.ManifestPath)
+
+	return nil
+}
+
+func init() {
+	RootCmd.AddCommand(exportCmd)
+	exportCmd.Flags().StringVarP(&utils.Flags.ProjectPath, "pathpath", "p", ".", "name of the managed project")
```
[the remark this answers is not in the document]
Comment (kpavel) [reply cmd/export.go]: Done. As well fixed above issue with rules export when target action is sequence.

**Comment by pritidesai** [comment]
LGTM, please merge. I have created issue for manifest formatting here:
https://github.com/apache/incubator-openwhisk-wskdeploy/issues/728

**Comment by csantanapr** [comment]
I really like this feature and would like to see it land but please cleaned up the output manifest file

**Comment by kpavel** [comment]
@csantanapr There an issue #728 @pritidesai opened for manifest formatting. It's not directly related to export, as export stores the manifest in same format existing `wskdeploy init` or `wskdeploy add` do.

**Comment by csantanapr** [comment]
@kpavel thanks for the clarification and again thanks for the contribution 💯

**Comment by csantanapr** [comment]
@mrutkows @pritidesai This would be a good feature to have looking forward to try it out once is merged.

**Comment by mrutkows** [comment]
@pritidesai @kpavel Would like to remove empty maps and interfaces from what the yaml parser generates from our schema.  We can look at in a separate issue; want to add this function.

**Comment by mrutkows** [comment]
@kpavel have you signed the Apache ICLA?

**Pull request merged**